### PR TITLE
[Fix](Row store) fix row store with invalid json string in variant type

### PR DESCRIPTION
**Pull request by eldenmoon**
Previous we allow invalid text as variant in PR #37794 and store as string type.But in encoding rowstore we CHECK the json is valid and store as jsonb binary field.In this PR we support the invalid json encoding as row store

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Review by xiaokang — DISMISSED** [review]
LGTM

File: be/src/vec/data_types/serde/data_type_object_serde.cpp [hunk 1]
```diff
@@ -98,8 +98,13 @@ void DataTypeObjectSerDe::write_one_cell_to_jsonb(const IColumn& column, JsonbWr
     JsonbParser json_parser;
     // encode as jsonb
     bool succ = json_parser.parse(value_str.data(), value_str.size());
-    // maybe more graceful, it is ok to check here since data could be parsed
-    CHECK(succ);
+    if (!succ) {
```
Comment (xiaokang) [inline be/src/vec/data_types/serde/data_type_object_serde.cpp:101]: if (succ) {
  // write binary
} else {
  // write string
}

Comment (eldenmoon) [reply be/src/vec/data_types/serde/data_type_object_serde.cpp:101]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**